polyfill.js

```diff
@@ -37,5 +37,5 @@ if (typeof Promise.prototype.finally !== 'function') {
 			return newPromise;
 		}
 	};
-	Promise.prototype.finally = shim.finally;
```

Comment:
This should conditionally detect `Object.defineProperty`, and fall back to assignment in ES3 browsers. Let's also explicitly set `enumerable` to `false`.